### PR TITLE
Add Magic areas 2

### LINKED ISSUE #16: Room button badges/actions
Room buttons:
show temp / humid 

Click actions for room buttons
click : go to room (same as now)
long click : toggle light group for room
double click :  more info light group ( dimming color control diag)

Room button 
badge icon:  
door if door is open  (if room has door)
lock if lock is unlocked ( if room has lock)
window if window is open (if room has window)
fan icon if fan is on (if room has fan)
motion if motion detected. ( if room has motion sensor)

badge color:
red if lock
yellow if door
blue if window
blue if fan 
green if motion 

image

Gives a quick overview of WHICH doors, windows, fans, motion sensors are active.

### PR DESCRIPTION
**Pull request by adrienbrault**
Hey,

This PR contains https://github.com/AalianKhan/mushroom-strategy/pull/44 rebased, along with some refactoring based on @DigiLive's [comment](https://github.com/AalianKhan/mushroom-strategy/pull/44#pullrequestreview-1633369922).

Here's what it looks like on my dashboard:

### CODE REVIEW

**Comment by adrienbrault** [comment]
@DigiLive I've updated the branch with your suggestions.

I think the configuration format should be improved though

**Comment by DigiLive** [comment]
> I think the configuration format should be improved though

Which format of which configuration do you suggest to what? Do you have a code example?

File: src/Helper.js [hunk 1]
```diff
@@ -355,7 +356,15 @@ class Helper {
 
     // Get states whose entity-id starts with the given string.
     const stateEntities = Object.values(this.#hassStates).filter(
-        state => state.entity_id.startsWith(`${domain}.`),
+        state => {
+          if (deviceClass
```
Comment (DigiLive) [inline src/Helper.js:360]: Can the return conditions rewritten to a ternary operator statement?
E.g.: ``return deviceClass && state.attributes.device_class === deviceClass ? state.entity_id.startsWith(`${domain}.`) : false;``

File: src/Helper.js [hunk 2]
```diff
@@ -331,16 +331,17 @@ class Helper {
...
    * The result excludes hidden and disabled entities.
    *
    * @param {areaEntity} area Area entity.
    * @param {string} domain Domain of the entity-id.
+   * @param {string|null} deviceClass Device class attribute of the state.
    *
    * @return {stateObject[]} Array of state entities.
    */
-  static getStateEntities(area, domain) {
+  static getStateEntities(area, domain, deviceClass = null) {
```
Comment (DigiLive) [inline src/Helper.js:344]: `deviceClass = null` or `deviceClass = undefined`?

File: src/Helper.js [hunk 3]
```diff
@@ -355,7 +356,15 @@ class Helper {
 
     // Get states whose entity-id starts with the given string.
     const stateEntities = Object.values(this.#hassStates).filter(
-        state => state.entity_id.startsWith(`${domain}.`),
+        state => {
+          if (deviceClass
+            && state.attributes.device_class !== deviceClass
```
Comment (DigiLive) [inline src/Helper.js:361]: Unresolved variable device_class.

It should be included in typedef `stateObject` in file `src/typedefs.js`

File: src/cards/AreaCard.js [hunk 4]
```diff
@@ -57,6 +58,68 @@ class AreaCard extends AbstractCard {
...
+
+  /**
+   * @inheritdoc
+   */
+  getCard() {
+    let card = super.getCard();
+
+    if (!card.secondary) {
+      // Get or determine the relevant sensor entity IDs based on options or default behavior
+      const temperatureSensorId = this.options?.temperature || Helper.getStateEntities(this.entity, "sensor", "temperature")[0]?.entity_id;
```
Comment (DigiLive) [inline src/cards/AreaCard.js:70]: The type of field `options` in `AbstractClass` should change to `{abstractOptions & Object<string, any>}` to allow unknown additional properties.

Applies to lines #L70, #L71, #L72, #L92, #L93, #L94.

Comment (DigiLive) [inline src/cards/AreaCard.js:70]: Argument type  `hassEntity | areaEntity` is not assignable to parameter type `areaEntity`.

Maybe it's more proper to declare private field `#entity `in `AreaCard` as type `areaEntity` and assign in the value of parameter `area` in the constructor. And then use `this.#entity` in the function call of `Helper.getStateEntities`

Applies to lines #L70, #L71, #L72, #L92, #L93, #L94.

**Review by DigiLive — CHANGES_REQUESTED** [review]

File: src/Helper.js [hunk 5]
```diff
@@ -331,16 +331,17 @@ class Helper {
...
 
   /**
-   * Get state entities, filtered by area and domain.
+   * Get state entities, filtered by area, domain and optionally deviceClass.
    *
    * The result excludes hidden and disabled entities.
    *
    * @param {areaEntity} area Area entity.
    * @param {string} domain Domain of the entity-id.
+   * @param {string|null} deviceClass Device class attribute of the state.
```
Comment (DigiLive) [inline src/Helper.js:340]: `deviceClass` should be `[deviceClass]` if it's an optional parameter.

**Comment by DigiLive** [comment]
Please remove `./dist/mushroom-strategy.js` from your commits.
It will create a merge conflict. The file will be build at GitHub when an applicable PR is merged into the `main branch`.

**Comment by DigiLive** [comment]
> [!WARNING]
> The `main` branch has been switched to TypeScript.
> Update your feature branch accordingly!

**Pull request closed without merging**